src/node/iir_filter.rs

```diff
@@ -0,0 +1,652 @@
+            5.051_316e-3,
+            1.130_323_5e-2,
+            2.230_340_2e-2,
+            4.311_698e-2,
+            8.843_45e-2,
+            2.146_620_2e-1,
+            6.802_952e-1,
+        ];
+        let context = OfflineAudioContext::new(2, LENGTH, SampleRate(44_100));
+        let feedforward = vec![
+            0.019_618_022_238_052_212,
+            -0.036_007_928_102_449_24,
+            0.019_618_022_238_052_21,
+        ];
+        let feedback = vec![1., 1.576_436_200_538_313_7, 0.651_680_173_116_867_3];
+
+        let options = IirFilterOptions {
+            feedback,
+            feedforward,
+            channel_config: ChannelConfigOptions::default(),
+        };
+        let iir = IirFilterNode::new(&context, options);
+
+        let mut frequency_hz = [
+            0., 2205., 4410., 6615., 8820., 11025., 13230., 15435., 17640., 19845.,
+        ];
+        let mut mag_response = [0.; 10];
+        let mut phase_response = [0.; 10];
+
+        iir.get_frequency_response(&mut frequency_hz, &mut mag_response, &mut phase_response);
+
+        assert_float_eq!(mag_response, ref_mag, ulps_all <= 0);
+    }
+
+    #[test]
+    fn default_periodic_wave_rendering_should_match_snapshot() {
+        // the snapshot data has been verified by fft to make sure that the frequency
+        // response correspond to a HP filter with Fc 4000 Hz
+        let ref_filtered =
+            snapshot::read("./snapshots/white_hp.json").expect("Reading snapshot file failed");
+
+        let mut context = OfflineAudioContext::new(1, LENGTH, SampleRate(44_100));
+
+        // setup background music:
+        // read from local file
+        let file = File::open("white.ogg").unwrap();
+        // decode file to media stream
+        let stream = OggVorbisDecoder::try_new(file).unwrap();
+        // wrap stream in MediaElement, so we can control it (loop, play/pause)
+        let media = MediaElement::new(stream);
+        // register as media element in the audio context
+        let background = context.create_media_element_source(media);
+
+        let feedforward = vec![
+            0.019_618_022_238_052_212,
+            -0.036_007_928_102_449_24,
+            0.019_618_022_238_052_21,
+        ];
+        let feedback = vec![1., 1.576_436_200_538_313_7, 0.651_680_173_116_867_3];
+
+        let options = IirFilterOptions {
+            feedback,
+            feedforward,
+            channel_config: ChannelConfigOptions::default(),
+        };
+        let iir = IirFilterNode::new(&context, options);
+
+        background.connect(&iir);
+        iir.connect(&context.destination());
+
+        background.start();
+        let output = context.start_rendering();
+
+        // retrieve processed data by removing silence chunk
+        // These silence slices are inserted inconsistently from an test execution to another
+        // Without these processing the test would be brittle
+        let data_ch: Vec<f32> = output
```

Comment:
I retry the test a bunch of time with a constant success.

Reply:
I think the inconsistencies come from the fact that you have used a MediaElement, which will perform some buffering that can be non-deterministic in tests. Can you try to use `create_media_stream_source` on the stream? That fixes the problem I hope and gives a bit more confidence in the test

Reply:
i will try your proposed alternative tomorrow

Reply:
> Can you try to use create_media_stream_source on the stream?

I tried with `create_media_stream_source` and it also fails from time to time if i don't add the counter measures of the actual test code

Reply:
That just sounds really strange to me.... But probably unrelated to your work. Let's get it merged